The Globe and Mail.js

```diff
@@ -1,223 +1,189 @@
-		}
-		
-		for (var i = 0; i < tagsContent.length; i++) {
-		     	if (tagsContent[i] != (" ") && tagsContent[i] != ("")) {
-			     	newItem.tags[i] = tagsContent[i];
-		     	}
-	     	}	
-		
-		associateMeta (newItem, metaTags, "headline", "title");
-		associateMeta (newItem, metaTags, "summary", "abstractNote");
-		associateMeta (newItem, metaTags, "desk", "section");	
-		associateMeta (newItem, metaTags, "article_id", "callNumber");
-		associateMeta (newItem, metaTags, "credit", "rights");
-	
-	//rest of author (shared between both newspaperArticle types)
-		if (author) {
-			
-		
-			if (author.substr(0,3).toLowerCase() == "by ") {
-				author= author.substr(3);
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
 			}
-		
-			var authors = author.toLowerCase().split(" and ");
-			for (var i=0; i<authors.length; i++) {
-				var author = authors[i];
-				var words = author.split(" ");
-				
-				for (var j=0; j<words.length; j++) {
-					if (words[j] != "") {
-						words[j] = words[j][0].toUpperCase() + words[j].substr(1).toLowerCase();
-					}
-				}
-				author = words.join(" ");
-				newItem.creators.push(Zotero.Utilities.cleanAuthor(author, "author"));			
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
 			}
-		}
-	}	
-	
-	newItem.url = doc.location.href;
-	newItem.complete();
-	
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
 }
 
-function doWeb(doc, url) {
-	var namespace = doc.documentElement.namespaceURI;
-	var nsResolver = namespace ? function(prefix) {
-		if (prefix == 'x') return namespace; else return null;
-	} : null;
+function scrape(doc, url) {
 	
-	var articles = new Array();
+	newItem = new Zotero.Item("newspaperArticle");
+	newItem.url = url;
+
+	//get headline
+	var title = ZU.xpathText(doc, '//article/h1');
+	if (!title) title = ZU.xpathText(doc, '//meta[@property="og:title"]/@content');
+	newItem.title = title;
+
+	//get abstract
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@property="og:description"]/@content');
 	
-	if (detectWeb(doc, url) == "multiple") {
-		var items = new Object();
-		var next_title = new Array();
-		
-		var titlesCount = doc.evaluate('count (//h3[@class="storyLink"]/a)', doc, nsResolver, XPathResult.ANY_TYPE, null);
-		var titles = doc.evaluate('//h3[@class="storyLink"]/a', doc, nsResolver, XPathResult.ANY_TYPE, null);		
-		
-		for (i=0; i < titlesCount.numberValue; i++) {
-			next_title = titles.iterateNext();
-			
-			if (next_title.href.match("story")) {
-				items[next_title.href] = next_title.textContent;
+	//get date
+	var xpathdate = '//time[@itemprop="datePublished"]';
+	var date = ZU.xpathText(doc, xpathdate);
+	if (date) {
+		newItem.date = ZU.strToISO(date);
+	}
+
+	//get author
+	var authors = ZU.xpath(doc, '//meta[@itemprop="author"]/@content');
+	for (var i in authors){
```

Comment:
`ZU.xpath` return an array. You always want to loop through arrays sequentially, i.e. `for (var i = 0; i<authors.length; i++)` both to avoid odd behavior and for efficiency.